### ISSUE TITLE
Doesn't start in Linux 

### ISSUE THREAD

**Issue by trapd00r** [jcs/sdorfehs#18]
sdorfehs doesn't start on Manjaro 5.14.

Tried with no config, and when starting an xterm from xinitrc and using gdb, this is the result:

```
(gdb) run
Starting program: /usr/local/bin/sdorfehs 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x0000555555564e26 in ?? ()
```
dmesg after segfault:
```
[ 1129.630927] resource sanity check: requesting [mem 0x000c0000-0x000fffff], which spans more than PCI Bus 0000:00 [mem 0x000d0000-0x000d3fff window]
[ 1129.630933] caller _nv000722rm+0x1ad/0x200 [nvidia] mapping multiple BARs
[ 1131.603428] sdorfehs[14049]: segfault at 4 ip 00005624cadafe26 sp 00007fff39fd57a0 error 4 in sdorfehs[5624cada3000+1e000]
[ 1131.603436] Code: 8b 73 18 48 89 05 8a d1 01 00 48 8b 05 4b d4 01 00 48 8b 90 98 00 00 00 48 63 87 e0 00 00 00 48 c1 e0 07 48 03 87 e8 00 00 00 <8b> 4a 04 44 8b 40 38 03 0a 8b 93 80 00 00 00 01 c9 e8 84 3c ff ff
[ 1131.603459] audit: type=1701 audit(1628154308.689:644): auid=1000 uid=1000 gid=1000 ses=2 pid=14049 comm="sdorfehs" exe="/usr/local/bin/sdorfehs" sig=11 res=1



```

**Comment by jcs** [comment]
`make clean` and then uncomment the `CFLAGS += -g` in `Makefile`, then `make` again and run that version with debugging symbols.

**Comment by trapd00r** [comment]
```
sdorfehs:globals.c:618: debug: Registered Atom 460 = _NET_SUPPORTED
sdorfehs:globals.c:618: debug: Registered Atom 437 = WM_CHANGE_STATE
sdorfehs:globals.c:618: debug: Registered Atom 542 = WM_COLORMAP_WINDOWS
sdorfehs:globals.c:618: debug: Registered Atom 371 = WM_DELETE_WINDOW
sdorfehs:globals.c:618: debug: Registered Atom 39 = WM_NAME
sdorfehs:globals.c:618: debug: Registered Atom 428 = WM_PROTOCOLS
sdorfehs:globals.c:618: debug: Registered Atom 436 = WM_STATE
sdorfehs:globals.c:618: debug: Registered Atom 429 = WM_TAKE_FOCUS
sdorfehs:globals.c:618: debug: Registered Atom 493 = _NET_ACTIVE_WINDOW
sdorfehs:globals.c:618: debug: Registered Atom 543 = _NET_CLIENT_LIST
sdorfehs:globals.c:618: debug: Registered Atom 544 = _NET_CLIENT_LIST_STACKING
sdorfehs:globals.c:618: debug: Registered Atom 545 = _NET_CURRENT_DESKTOP
sdorfehs:globals.c:618: debug: Registered Atom 546 = _NET_NUMBER_OF_DESKTOPS
sdorfehs:globals.c:618: debug: Registered Atom 462 = _NET_WORKAREA
sdorfehs:globals.c:618: debug: Registered Atom 367 = _NET_WM_NAME
sdorfehs:globals.c:618: debug: Registered Atom 346 = _NET_WM_PID
sdorfehs:globals.c:618: debug: Registered Atom 347 = _NET_WM_STATE
sdorfehs:globals.c:618: debug: Registered Atom 349 = _NET_WM_STATE_FULLSCREEN
sdorfehs:globals.c:618: debug: Registered Atom 351 = _NET_WM_WINDOW_TYPE
sdorfehs:globals.c:618: debug: Registered Atom 353 = _NET_WM_WINDOW_TYPE_DIALOG
sdorfehs:globals.c:618: debug: Registered Atom 478 = _NET_WM_WINDOW_TYPE_DOCK
sdorfehs:screen.c:396: debug: display string: DISPLAY=:0.0
sdorfehs:screen.c:396: debug: display string: DISPLAY=:0.0
sdorfehs:screen.c:396: debug: display string: DISPLAY=:0.0
sdorfehs:screen.c:224: debug: Xrandr primary screen 1431862624 detected
sdorfehs:actions.c:737: debug: realloc()ed key_table 2
sdorfehs:actions.c:737: debug: realloc()ed key_table 4
sdorfehs:actions.c:737: debug: realloc()ed key_table 8
sdorfehs:actions.c:737: debug: realloc()ed key_table 16
sdorfehs:actions.c:737: debug: realloc()ed key_table 32
sdorfehs:actions.c:737: debug: realloc()ed key_table 64
sdorfehs:actions.c:737: debug: realloc()ed key_table 128
sdorfehs:history.c:142: debug: could not load history from /home/scp1/.config/sdorfehs/history: No such file or directory
Program received signal SIGSEGV, Segmentation fault.
0x00005555555660fd in init_bar () at bar.c:130
130                 FONT_HEIGHT(rp_current_screen) * 2,
(gdb) Quit

```

**Comment by trapd00r** [comment]
I am using a triplehead setup, like this: [][  ][]. I disabled the non-primary ones but to no avail.

**Comment by jcs** [comment]
Can you update and try now?  I think this was fixed by 1c02f39b5f0ac30349f06c5fa68ae573dc1e9b06.

**Comment by trapd00r** [comment]
It was indeed! Thank you!
